File: timecop/php/lib/Entry.php
<?php

class Entry
{
	public $errors = Array();
	public $details = Array();
	protected $enteredAt;
	protected $valid;

	public function __construct($valid, $details=Array(), $errors=Array(), $enteredAt=null)
	{
		if (is_null($enteredAt)) $enteredAt = time();
		$this->enteredAt = $enteredAt;

		$this->valid = (boolean) $valid;
		$this->details = $details;
		$this->errors = $errors;
	}

	public function isValid()
	{
		return $this->valid;
	}

	public function getEntryDate()
	{
		return $this->enteredAt;
	}
}

File: timecop/ruby/lib/entry.rb
class Entry
  attr_accessor :errors, :details
  attr_reader :entered_at

  def initialize(valid, details={}, errors=[], entered_at=Time.now)
    @entered_at  = entered_at
    @valid       = !!valid
    self.details = details
    self.errors  = errors
  end

  def valid?
    @valid
  end
end
